src/main/java/net/minecraftforge/common/block/spreader/SpreadBehaviors.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.block.spreader;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraftforge.common.extensions.IForgeSpreadingBlock;
+import net.minecraftforge.registries.IRegistryDelegate;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpreadBehaviors
+{
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Map<IRegistryDelegate<Block>, Map<SpreaderType, ISpreadingBehavior>> SPREADERS = new HashMap<>();
```

Comment:
I feel like a [Guava `Table<IRegistryDelegate<Block>, SpreaderType, ISpreadingBehavior>`](https://github.com/google/guava/wiki/NewCollectionTypesExplained#table) would be cleaner here than a `Map` of `Map`s.